nixos/modules/virtualisation/containers.nix

```diff
@@ -463,10 +463,15 @@ in
                 A specification of the desired configuration of this
                 container, as a NixOS module.
               '';
-              type = lib.mkOptionType {
+              type = let
+                confPkgs = if config.pkgs == null then pkgs else config.pkgs;
```

Comment:
To get rid of the `nullOr` type (including needed conditionals), how about setting `containers.<name>.pkgs` to `pkgs` by default?

Reply:
I did this at first but it ofborg re-evaluate all packages.